### PR TITLE
docs: Provide API docs pattern, docs for fclasses and vital-flowcontainer

### PR DESCRIPTION
**Pull request by wodenx**
## Chnges
Implement typedoc patterns for fclasses and vital-fowcontainer
document best practices.

### CODE REVIEW

**Review by lucas-varela — CHANGES_REQUESTED** [review]
I got some comments, please take a look.

File: sites/__vital__/package.json [hunk 1]
```diff
@@ -72,8 +72,10 @@
     "clean": "gatsby clean",
     "dev-backend": "bodiless-backend",
     "dev-frontend": "gatsby develop",
+    "docs": "run-s build:docs serve:docs",
     "prepare": "node ../../scripts/set-missing-symlinks.js @bodiless",
-    "prestart": "run-s build:env-vars build:doc",
+    "prestart": "run-s build:env-vars build:docs",
+    "search-index": "create-search-index",
```
[the remark this answers is not in the document]
Comment (wodenx) [reply sites/__vital__/package.json:78]: i replaced it with "build:search-index"

File: packages/bodiless-ui/package.json [hunk 2]
```diff
@@ -25,12 +25,12 @@
     "access": "public"
   },
   "scripts": {
-    "build": "tsc -p ./tsconfig.json && npm run build:css && npm run copy && npm run build:api-doc",
+    "build": "npm run build:lib && npm run build:css && npm run copy",
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/bodiless-ui/package.json:28]: well ... we should standardize, i guess, but is there any advantage to run-s vs this pattern?

File: packages/bodiless-documentation/doc/Development/ApiDocumentationStandards.md [hunk 3]
```diff
@@ -150,36 +150,142 @@ BasicList(options: ListProps): Element
...
+  - `@category Utility`
+- For every token collection you create, define an `Interface` and document its
+  members. Ensure that your actual token collection object is typed using this
+  interface (that way you will ensure that there is a doc for every new token
+  you add).
+- For every clean component you create, define an `Interface` which defines
+  its designable components.  Document the purpose of each slot:
+  ```ts
+  /**
+   * Designable components fpr FooClean.
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/bodiless-documentation/doc/Development/ApiDocumentationStandards.md:246]: fixed.

File: packages/vital-flowcontainer/src/components/FlowContainer/index.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 /**
+ * teeth
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/vital-flowcontainer/src/components/FlowContainer/index.ts:2]: shrug.  removed.

File: typedoc.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  // readme: './API.md',
+  // categorizeByGroup: true,
```
[the remark this answers is not in the document]
Comment (wodenx) [reply typedoc.js:3]: i just removed it - turns out we don't need any global options atm.

File: packages/bodiless-ui/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucas-varela) [reply packages/bodiless-ui/package.json:28]: Using `run-s` wouldn't change the current behavior, just make it look more like the other files. Either way works great.

File: packages/fclasses/src/addClasses.tsx [hunk 6]
```diff
@@ -117,13 +175,29 @@ type ForwardRefProps = {
...
+ * specified operations will be applied in reverse order up the component tree,
+ * so that the outermost operations take precedence.
+ *
+ * Note, when making HTML tags stylable, you should include a type parameter
+ * to ensure that the resuting
+ *
+ * Adds [[StylableProps]] to the component.
+ *
+ * > When making a plain HTML tag stylable, use a type parameter to ensure
+ * > the resulting component has the correct prop types.
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/fclasses/src/addClasses.tsx:189]: fixed.

File: packages/fclasses/src/addClasses.tsx [hunk 7]
```diff
@@ -58,34 +91,59 @@ const modifyClassesIf = (operation: 'add' | 'remove') => <A extends object>(
...
- * Allows to add classes to a component conditionally.
+ * Add classes to a component conditionally.
+ *
+ * @param
+ * A custom condition hook which will be evaluated at render time
+ * to determine whether the classes should be added. You may use
+ * React hooks.
+ *
+ * @returns
+ * A HOC factory function wit the same signature as [[addClasses]].
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/fclasses/src/addClasses.tsx:102]: fixed.

File: packages/bodiless-layouts/src/FlowContainer/types.ts [hunk 8]
```diff
@@ -87,7 +87,21 @@ export type FlowContainerWrapperProps = {
...
+   * the layout of its children.
+   */
   Wrapper: ComponentOrTag<any>,
+  /**
+   * The wrapper for each component placed in the flow container.  Usually a `Div`.
+   * Width classes are applied to this wrapper dynamically to control layout.
+   */
   ComponentWrapper: ComponentOrTag<any>,
-};
+}
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/bodiless-layouts/src/FlowContainer/types.ts:107]: no semi needed for interface - this is changed bc before it was a tuype alias.

File: packages/bodiless-layouts/src/FlowContainer/index.tsx [hunk 9]
```diff
@@ -1,4 +1,5 @@
 /**
+ * Aaaaagh
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/bodiless-layouts/src/FlowContainer/index.tsx:2]: again (shrug). fixed.

File: packages/bodiless-ui/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/bodiless-ui/package.json:28]: leaving as is for now

File: packages/fclasses/src/Design.tsx [hunk 10]
```diff
@@ -31,8 +31,15 @@ import { withTransformer } from './Transformer';
 import { as, extendDesign } from './tokenSpec';
 
 /**
- * is an HOC that will attach a displayName to an object
- * @param name the name of the displayName.
+ * Creates an HOC that will attach a displayName to an object.
+ *
+ * @param name
+ * name the name of the displayName.
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/fclasses/src/Design.tsx:37]: fixed.

File: packages/fclasses/src/flowIf.tsx [hunk 11]
```diff
@@ -37,19 +39,66 @@ export const flowIf = <P extends object>(condition: Condition<P>): FlowHoc<P> =>
...
-
-export const and = <A extends object>(...funcs: FlowIfFunc<A>[]) => (props: A) => (
+/**
+ * Performs a logical AND of a series of [[Condition]]s.
+ *
+ * @param funcs
+ * The [[Condition]]s to AND together.
+ *
+ * @returns
+ * A [[Condition]] which evaluates to true of all of the
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/fclasses/src/flowIf.tsx:49]: fixed.

File: packages/fclasses/src/flowIf.tsx [hunk 12]
```diff
@@ -37,19 +39,66 @@ export const flowIf = <P extends object>(condition: Condition<P>): FlowHoc<P> =>
...
-export const or = <A extends object>(...funcs: FlowIfFunc<A>[]) => (props: A) => (
+
+/**
+ * Performs a logical OR of a series of [[Condition]]s.
+ *
+ * @param funcs
+ * The [[Condition]]s to OR together.
+ *
+ * @returns
+ * A [[Condition]] which evaluates to rue of any of the
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/fclasses/src/flowIf.tsx:65]: fixed.

File: packages/fclasses/src/flowIf.tsx [hunk 13]
```diff
@@ -37,19 +39,66 @@ export const flowIf = <P extends object>(condition: Condition<P>): FlowHoc<P> =>
...
 
-export const not = <A extends object>(...funcs: FlowIfFunc<A>[]) => (props: A) => (
+/**
+ * Negates one or more [[Condition]].
+ *
+ * @param funcs
+ * The [[Condition]]s to negate
+ *
+ * @returns
+ * A [[Condition]] which evaluates to rue if all the
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/fclasses/src/flowIf.tsx:81]: fixed.

**Review by lucas-varela — APPROVED** [review]

**Review by lucas-varela — APPROVED** [review]

**Review by hvanyo — APPROVED** [review]

**Review by hvanyo — APPROVED** [review]

**Review by lucas-varela — APPROVED** [review]

**Pull request merged**